src/main/java/com/commercetools/project/sync/state/StateSyncer.java

```diff
@@ -78,17 +77,7 @@ public static StateSyncer of(
   @Nonnull
   @Override
   protected CompletionStage<List<StateDraft>> transform(@Nonnull List<State> states) {
-    return toStateDrafts(getSourceClient(), referenceIdToKeyCache, states)
-        .handle(
-            (stateDrafts, throwable) -> {
-              if (throwable != null) {
-                if (LOGGER.isWarnEnabled()) {
-                  LOGGER.warn(throwable.getMessage(), getCompletionExceptionCause(throwable));
-                }
-                return List.of();
-              }
-              return stateDrafts;
-            });
+    return toStateDrafts(getSourceClient(), referenceIdToKeyCache, states);
```

Comment:
I see similar codes also in these places: https://github.com/commercetools/commercetools-project-sync/blob/3a5df41f88284510db3b1548d9690a325e2a5996/src/main/java/com/commercetools/project/sync/product/ProductSyncer.java#L117, https://github.com/commercetools/commercetools-project-sync/blob/3a5df41f88284510db3b1548d9690a325e2a5996/src/main/java/com/commercetools/project/sync/type/TypeSyncer.java#L102. Please check if they could be simplified as well.

Reply:
I've simplified only for the TypeSyncer: [970c8ac](https://github.com/commercetools/commercetools-project-sync/pull/531/commits/970c8ac5310acf688a58a784a7f6c212a373d035)
The ProductSync throws a `ReferenceResolutionException` which needs to be handled at this place, so this is correct behaviour.